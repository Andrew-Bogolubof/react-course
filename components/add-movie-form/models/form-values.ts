export type FormValues = {
  title: string;
  releaseDate: string;
  movieUrl: string;
  genres: string[];
  overview: string;
  runtime: number | string;
};
