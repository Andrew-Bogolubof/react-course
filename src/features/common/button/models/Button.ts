export enum Button {
  Primary = 'Primary',
  Secondary = 'Secondary',
  Empty = 'Empty',
  Cancel = 'Cancel',
  Close = 'Close',
  CloseSmall = 'CloseSmall',
}
