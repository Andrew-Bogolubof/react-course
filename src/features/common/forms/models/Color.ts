export enum TextColor {
  White = 'white',
  Gray = 'gray',
}
