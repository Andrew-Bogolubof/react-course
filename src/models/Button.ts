export enum Button {
  Primary = 'Primary',
  Secondary = 'Secondary',
  Cancel = 'Cancel',
}
