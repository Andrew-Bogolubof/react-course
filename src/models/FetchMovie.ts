import { Movie } from './Movie';

export enum SortOrder {
  DESC = 'desc',
  ASC = 'asc',
}
export enum SortBy {
  TITLE = 'title',
  GENRES = 'genres',
}

export type FetchMovieQuery = {
  sortBy?: keyof Movie;
  sortOrder?: SortOrder;
  search?: string;
  searchBy?: SortBy;
  filter?: string[];
  offset?: string;
  limit?: string;
};
