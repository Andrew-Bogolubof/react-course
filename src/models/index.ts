export * from './Movie';
export * from './FetchMovie';
