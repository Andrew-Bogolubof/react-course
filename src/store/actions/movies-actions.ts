import { createAction } from 'redux-actions';
import { FetchMovieQuery, Movie } from '../../models';

export enum MoviesActions {
  SET_MOVIES = 'SET_MOVIES',
  FETCH_MOVIES = 'FETCH_MOVIES',
  FETCH_MOVIE = 'FETCH_MOVIE',
  CREATE_MOVIE = 'CREATE_MOVIE',
  UPDATE_MOVIE = 'UPDATE_MOVIE',
  SET_SEARCH_QUERY = 'SET_SEARCH_QUERY',
  DELETE_MOVIE = 'DELETE_MOVIE',
}

export const setMovies = createAction<Movie[]>(MoviesActions.SET_MOVIES);
export type SetMovies = ReturnType<typeof setMovies>;

export const fetchMovies = createAction<FetchMovieQuery>(MoviesActions.FETCH_MOVIES);
export type FetchMovies = ReturnType<typeof fetchMovies>;

export const fetchMovie = createAction<{ id: number }>(MoviesActions.FETCH_MOVIE);
export type FetchMovie = ReturnType<typeof fetchMovie>;

export const createMovie = createAction<
  Pick<Movie, 'title' | 'release_date' | 'poster_path' | 'genres' | 'overview' | 'runtime'>
>(MoviesActions.CREATE_MOVIE);
export type CreateMovie = ReturnType<typeof createMovie>;

export const updateMovie = createAction<Omit<Partial<Movie>, 'id'> & { id: number }>(
  MoviesActions.UPDATE_MOVIE
);
export type UpdateMovie = ReturnType<typeof updateMovie>;

export const deleteMovie = createAction<{ id: Movie['id'] }>(MoviesActions.DELETE_MOVIE);
export type DeleteMovie = ReturnType<typeof deleteMovie>;
