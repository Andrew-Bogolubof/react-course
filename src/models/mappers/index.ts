import { SortBy, SortByField } from '../FetchMovie';

export const mapSortByToSortByField = (sortBy: SortBy | undefined): SortByField => {
  switch (sortBy) {
    case SortBy.RATING:
      return SortByField.RATING;
    case SortBy.RELEASE_DATE:
      return SortByField.RELEASE_DATE;
    case SortBy.TITLE:
      return SortByField.TITLE;
    default:
      return SortByField.TITLE;
  }
};

export const mapSortByFieldToSortBy = (sortBy: SortByField | undefined): SortBy => {
  switch (sortBy) {
    case SortByField.RATING:
      return SortBy.RATING;
    case SortByField.RELEASE_DATE:
      return SortBy.RELEASE_DATE;
    case SortByField.TITLE:
      return SortBy.TITLE;
    default:
      return SortBy.TITLE;
  }
};
