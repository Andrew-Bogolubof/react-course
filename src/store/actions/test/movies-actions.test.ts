import { Movie } from '../../../models';
import { createMovie, deleteMovie, fetchMovies, setMovies, updateMovie } from '../movies-actions';
import { mockMovie, mockQuery } from './fixtures';

describe('Movies actions', () => {
  it('should correctly create setMovies action', () => {
    const movies = [] as Movie[];

    const result = setMovies(movies);

    expect(result).toMatchSnapshot();
  });

  it('should correctly create createMovie action', () => {
    const movie = mockMovie;

    const result = createMovie(movie);

    expect(result).toMatchSnapshot();
  });

  it('should correctly create updateMovie action', () => {
    const movie = mockMovie;

    const result = updateMovie(movie);

    expect(result).toMatchSnapshot();
  });

  it('should correctly create deleteMovie action', () => {
    const id = { id: 1 };

    const result = deleteMovie(id);

    expect(result).toMatchSnapshot();
  });

  it('should correctly create fetchMovies action', () => {
    const result = fetchMovies(mockQuery);

    expect(result).toMatchSnapshot();
  });
});
