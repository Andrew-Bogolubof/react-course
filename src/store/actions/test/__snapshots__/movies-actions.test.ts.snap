// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies actions should correctly create createMovie action 1`] = `
Object {
  "payload": Object {
    "budget": 200000000,
    "genres": Array [
      "Fantasy",
      "Adventure",
      "Science Fiction",
    ],
    "id": 181808,
    "overview": "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
    "poster_path": "https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
    "release_date": "2017-12-13",
    "revenue": 1325937250,
    "runtime": 152,
    "tagline": "The Saga Continues",
    "title": "Star Wars: The Last Jedi",
    "vote_average": 7.1,
    "vote_count": 4732,
  },
  "type": "CREATE_MOVIE",
}
`;

exports[`Movies actions should correctly create deleteMovie action 1`] = `
Object {
  "payload": Object {
    "id": 1,
  },
  "type": "DELETE_MOVIE",
}
`;

exports[`Movies actions should correctly create fetchMovies action 1`] = `
Object {
  "payload": Object {
    "filter": Array [],
    "limit": "6",
    "offset": "6",
    "search": "My Search",
    "searchBy": "title",
    "sortBy": "vote_count",
    "sortOrder": "desc",
  },
  "type": "FETCH_MOVIES",
}
`;

exports[`Movies actions should correctly create setMovies action 1`] = `
Object {
  "payload": Array [],
  "type": "SET_MOVIES",
}
`;

exports[`Movies actions should correctly create updateMovie action 1`] = `
Object {
  "payload": Object {
    "budget": 200000000,
    "genres": Array [
      "Fantasy",
      "Adventure",
      "Science Fiction",
    ],
    "id": 181808,
    "overview": "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
    "poster_path": "https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
    "release_date": "2017-12-13",
    "revenue": 1325937250,
    "runtime": 152,
    "tagline": "The Saga Continues",
    "title": "Star Wars: The Last Jedi",
    "vote_average": 7.1,
    "vote_count": 4732,
  },
  "type": "UPDATE_MOVIE",
}
`;
